docs/core/concepts/templating.md

```diff
@@ -0,0 +1,112 @@
+# Templating names
+
+There are several places within Prefect where names can be templated at runtime to create a value relevant to that specific run.
+These names can be passed as literal strings, e.g. `"my_foo_task"`,  but to get dynamic values based on the current state, we'll need to use _template_ strings.
+Here, we take advantage of Python's [format string](https://www.python.org/dev/peps/pep-3101/#format-strings) and provide various arguments.
+Instead of a static string, you'll pass a string that will be formatted at runtime, e.g. `"foo-{today}"`, where `today` is the name of one of the variables we provide.
+
+## Where can I use templating?
+
+Templating is available for:
+
+- [Task run names](/core/idioms/task-run-names.html)
+- [Task target paths](/core/idioms/targets.html)
+- [Task result locations](/core/concepts/results.html#templating-result-locations)
+
+## What variables can I use?
+
+All templatable objects are passed:
+
+- The [Parameters](/core/concepts/parameters.html) of the flow
+- The contents of the [Prefect context](/api/latest/utilities/context.html)
+- The inputs to the task
+
+::: warning Naming collisions
+
+The variables are loaded in the order shown above. Collisions will be resolved such that the last loaded value overwrites those above it. For example, if your a task input name is the same as the name of a variable in the context, the input variable value will overwrite context value.
+
+:::
+
+## Examples
+
+The name can be templatable string like `"{val}"`. Make sure not to try to format the string early, e.g. `f"{val}"` as this will be formatted when your flow is *defined* rather than when it is *run*.
```

Comment:
```suggestion
The name can be a templatable string like `"{val}"`. Make sure not to try to format the string early, e.g. `f"{val}"` as this will be formatted when your flow is *defined* rather than when it is *run*.
```